### PR TITLE
Change gc test to not depend on the default value.

### PR DESCRIPTION
**Pull request by mm-near**

### CODE REVIEW

**Comment by matklad** [comment]
One rule of thumb which ended up being useful for rust-analyzer is "never implement `Default` for Config structs". "Default application configuration" is application-level concern, not a language-level concern. 

Instead, there are two places in the codebase where various `Config`s are created:

* `main`, which implements application level logic for figuring out actual defaults
* default config for tests, which is a single constant/factory function.

I think for near such rule of thumb would also make sense, but we can only have it if we refactor the code to have only one or two points where we instantiate configs.

File: chain/chain/src/tests/gc.rs [hunk 1]
```diff
@@ -661,7 +663,14 @@ fn test_fork_far_away_from_epoch_end() {
 
     // GC execution
     chain1
-        .clear_data(tries1.clone(), &GCConfig { gc_blocks_limit: 100, ..GCConfig::default() })
+        .clear_data(
+            tries1.clone(),
+            &GCConfig {
+                gc_blocks_limit: 100,
+                gc_fork_clean_step: fork_clean_step,
+                ..GCConfig::default()
```
Comment (matklad) [inline chain/chain/src/tests/gc.rs:671]:
```suggestion
```

That way, if we add a new field to config, we'd need to think what value should the test use.

**Review by matklad — APPROVED** [review]

File: chain/chain/src/tests/gc.rs [hunk 2]
```diff
@@ -627,16 +627,18 @@ fn test_gc_star_large() {
...
     let simple_chains = vec![
         SimpleChain { from: 0, length: 5, is_removed: false },
         SimpleChain { from: 5, length: 2, is_removed: true },
         // We want the chain to end up exactly at the new epoch start.
-        SimpleChain { from: 5, length: 6600 - 5 + 1, is_removed: false },
+        SimpleChain { from: 5, length: 6 * epoch_length - 5 + 1, is_removed: false },
     ];
 
     let num_shards = 1;
     // We pick the epoch length that is greater than 1k (GC_FORK_CLEAN_STEP)
```
Comment (mina86) [inline chain/chain/src/tests/gc.rs:640]: This comment is no longer accurate.

**Review by mina86 — APPROVED** [review]

**Review by marcelo-gonzalez — APPROVED** [review]
looks like it needs a quick `cargo fmt` :)

**Pull request merged**